service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFSFilter.java

```diff
@@ -0,0 +1,132 @@
+    private static final String KEY_LESS = "lessThan";
+    private static final String KEY_LESS_OR_EQUAL = "atMost";
+
+    private static final String KEY_LIKE = "like";
+    private static final String KEY_NOT_LIKE = "notLike";
+    private static final String KEY_IN = "in";
+    private static final String KEY_NOT_IN = "notIn";
+
+    private static FilterFactory ff = CommonFactoryFinder.getFilterFactory();
+
+    protected static Filter getBBOXFilter (OskariLayer layer, ReferencedEnvelope bbox) {
+        WFSLayerCapabilities caps = new WFSLayerCapabilities(layer.getCapabilities());
+        String geomName = caps.getGeometryAttribute();
+        return ff.bbox(geomName,
+                bbox.getMinX(), bbox.getMinY(),
+                bbox.getMaxX(), bbox.getMaxY(),
+                CRS.toSRS(bbox.getCoordinateReferenceSystem()));
+    }
+
+    protected static Filter getFilter(JSONObject filterJson) {
+        Filter filter = null;
+        JSONObject prop = JSONHelper.getJSONObject(filterJson, KEY_PROPERTY);
+        if (prop != null) {
+            filter = getPropertyFilter(prop);
+        }
+        List <Filter> ands = JSONHelper.getArrayAsList(JSONHelper.getJSONArray(filterJson, KEY_AND))
+                .stream().map(f -> getPropertyFilter((JSONObject) f)).collect(Collectors.toList());
+        if (!ands.isEmpty()) {
+            filter = appendFilter(filter, ff.and(ands));
+        }
+        List <Filter> ors = JSONHelper.getArrayAsList(JSONHelper.getJSONArray(filterJson, KEY_OR))
+                .stream().map(f -> getPropertyFilter((JSONObject) f)).collect(Collectors.toList());
+        if (!ors.isEmpty()) {
+            filter = appendFilter(filter, ff.or(ors));
+        }
+
+        return filter;
+    }
+    private static Filter appendFilter(final Filter main, final Filter toAppend) {
+        if (main == null) {
+            return toAppend;
+        } else {
+            return ff.and(main, toAppend);
+        }
+    }
+    private static Filter getPropertyFilter(JSONObject property) {
+        String key = JSONHelper.optString(property, KEY);
+        boolean match = JSONHelper.getBooleanFromJSON(property,KEY_MATCH_CASE, true);
+        Expression name = ff.property(key);
+        if (property.has(KEY_EQUAL)) {
+            return ff.equal(name, ff.literal(JSONHelper.get(property, KEY_EQUAL)), match);
+        }
+        if (property.has(KEY_GREATER)) {
+            Filter greater = ff.greater(name, ff.literal(JSONHelper.get(property, KEY_GREATER)));
+            if (property.has(KEY_LESS)) {
+                Filter less = ff.less(name, ff.literal(JSONHelper.get(property, KEY_LESS)));
+                return ff.and(greater, less);
+            }
+            if (property.has(KEY_LESS_OR_EQUAL)) {
+                Filter lessEqual = ff.lessOrEqual(name, ff.literal(JSONHelper.get(property, KEY_LESS_OR_EQUAL)));
+                return ff.and(greater, lessEqual);
```

Comment:
`prop > x AND prop  <= x` is always false. Same value can't be greater than x and less than or equal to x at the same time.